stl-thumb: init at 0.5.0

SyntaxualSugar opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

stl-thumb is a thumbnail generator for STL and OBJ files (helpful for file managers). It is tested against Gnome and KDE (KDE with an extra plugin needed), and Windows, but any file manager should work.

[stl-thumb](https://github.com/unlimitedbacon/stl-thumb)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  pkgs,
```
```suggestion
```
We don't call `pkgs` inside Nixpkgs.

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  pkgs,
+  stdenv,
```
```suggestion
```
Please remove all unused inputs.

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+    --add-needed ${libGL}/lib/libEGL.so \
+    $out/bin/stl-thumb
+  '';
+  meta = with lib; {
```
```suggestion
  meta = {
```

Nested `with;` expressions is an anti-pattern.

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+  src = fetchFromGitHub {
+    owner = "unlimitedbacon";
+    repo = pname;
+    rev = "6bc90573b2cddda3f8916b2716898298475406c0";
```
```suggestion
    tag = "v${version}";
```

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+
+  src = fetchFromGitHub {
+    owner = "unlimitedbacon";
+    repo = pname;
```
```suggestion
    repo = "stl-thumb";
```
Avoid interpolating `pname`.

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+}:
+pkgs.rustPlatform.buildRustPackage rec {
+  pname = "stl-thumb";
+  version = "0.5.0-6bc9057";
```
```suggestion
  version = "0.5.0";
```

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+    sha256 = "sha256-G5zk1FR18rhWdC1S1cGOcsAHRpq03A7+ivNs4sUpl54=";
+  };
+
+  doCheck = false;
```
Please add a comment explaining why we're skipping tests.

SyntaxualSugar replied:
There are 3 tests in the project and all of them need an xdg runtime configured. I haven't been able to figure out how to get it working within the check phase.

lucasew replied:
If they only need a dir create a dir with `mktemp -d` then and pass it to a environment variable in preCheck

SigmaSquadron commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,62 @@
+    fontconfig
+    libX11
+  ];
+  propagatedBuildInputs = [
```
Why are we propagating these?

SyntaxualSugar replied:
I believe I was propagating them while trying to get the tests to work. For my education, what is the downside of propagating inputs?

SigmaSquadron reviewed (commented):
1. For consistency with other packages, consider running `nix-shell -p nixpkgs-hammering --run "nixpkgs-hammer stl-thumb"` in the root Nixpkgs directory, and make the edits the program tells you to.

2. Please review the [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) and squash your commits appropriately. There should be two commits total:
    1. `maintainers: add SyntaxualSugar`
    2. `stl-thumb: init at 0.5.0`

SyntaxualSugar commented:
> 1. For consistency with other packages, consider running `nix-shell -p nixpkgs-hammering --run "nixpkgs-hammer stl-thumb"` in the root Nixpkgs directory, and make the edits the program tells you to.
> 
>     2. Please review the [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) and squash your commits appropriately. There should be two commits total:
>        
>        1. `maintainers: add SyntaxualSugar`
>        2. `stl-thumb: init at 0.5.0`

Thank you for all of the feedback! I will clean up the changes and come back with a new pull request.

SigmaSquadron commented:
There's... really no need. Just force-push this one. It's best if we don't lose the review history.

SyntaxualSugar committed:
maintainers: add SyntaxualSugar

SyntaxualSugar committed:
stl-thumb: init at 0.5.0

tmp

lucasew commented:
the maintainers commit has a weird diff

lucasew commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 370942`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>stl-thumb</li>
  </ul>
</details>

SyntaxualSugar commented:
> the maintainers commit has a weird diff

This has been fixed. I think I had used the wrong nix fmt tool. I should have caught it before pushing.

lucasew commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,78 @@
+  src = fetchFromGitHub {
+    owner = "unlimitedbacon";
+    repo = "stl-thumb";
+    rev = "v${version}";
```
rev -> tag

lucasew commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,78 @@
+    sha256 = "sha256-sMgYrVQAtyVTfQyuTb/8OtRuDpagNQpt5YoF9lGIMHg=";
+  };
+
+  cargoLock = {
```
Use cargoHash instead

lucasew replied:
Oops, you added already so this part can be removed then

lucasew commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,78 @@
+    --add-needed ${lib.getLib libGL}/lib/libEGL.so \
+    $out/bin/stl-thumb
+  '';
+
```
passthru.updateScript = nix-update-script { }

SigmaSquadron replied:
This shouldn't be needed, as `r-ryantm` can retrieve updates from GitHub.

SyntaxualSugar replied:
Sorry, can you explain what you mean by this?

DaRacci commented on pkgs/by-name/st/stl-thumb/package.nix:
```diff
@@ -0,0 +1,78 @@
+    cp libstl_thumb.h $out/include
+    mkdir -p $out/thumbnailers
+    mkdir -p $out/mime/packages
+    cp stl-thumb.thumbnailer $out/thumbnailers/stl-thumb.thumbnailer
```
Shouldn't this be `$out/share/thumbnailers/stl-thumb.thumbnailer`?

marcusramberg replied:
```
nixos/modules/config/system-path.nix
172:      "/share/thumbnailers"
```
Looks like it should.

marcusramberg commented:
@SyntaxualSugar are you still working on this? Would be awesome to get stl-thumb into nixpkgs. I believe https://github.com/NixOS/nixpkgs/commit/b1b114e5027232f0d5faea7d9e1cba406f10c199.patch addresses the remaining review comments.

SyntaxualSugar commented:
> @SyntaxualSugar are you still working on this? Would be awesome to get stl-thumb into nixpkgs. I believe https://github.com/NixOS/nixpkgs/commit/b1b114e5027232f0d5faea7d9e1cba406f10c199.patch addresses the remaining review comments.

I was planning on continuing it, but I haven't had time lately. Feel free to pick it up if you do. I'd just like to be able to see it and stl-thumb-kde in nixpkgs, I won't be hurt if I'm not the person to get it there.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/buildrustpackage-installs-wrong-version-of-dependency-glium/77369/1
